packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -72,52 +72,95 @@ contract FederatedAttestations is
   }
 
   function _isRevoked(address signer, uint256 time) internal view returns (bool) {
-    if (revokedSigners[signer] > 0 && revokedSigners[signer] >= time) {
-      return true;
-    }
-    return false;
+    require(time >= 0);
+    uint256 revokedOn = revokedSigners[signer];
+    return revokedOn > 0 && revokedOn <= time ? true : false;
   }
 
+  /**
+   * @notice Returns info about attestations (with unrevoked signers)
+   *   for an identifier produced by a list of issuers
+   * @param identifier Hash of the identifier
+   * @param trustedIssuers Array of n issuers whose attestations will be included
+   * @param maxAttestations Limit the number of attestations that will be returned
+   * @return for m found attestations, m <= maxAttestations:
+   *         [
+   *           Array of m accounts,
+   *           Array of m issuedOns,
+   *           Array of m signers
+   *         ]; index corresponds to the same attestation
+   * @dev Adds attestation info to the arrays in order of provided trustedIssuers
+   * @dev Expectation that only one attestation exists per (identifier, issuer, account)
+   */
+
+  // TODO ASv2 consider also returning an array with counts of attestations per issuer
   function lookupAttestations(
     bytes32 identifier,
     address[] memory trustedIssuers,
     uint256 maxAttestations
-  ) public view returns (IdentifierOwnershipAttestation[] memory) {
+  ) public view returns (address[] memory, uint256[] memory, address[] memory) {
     // Cannot dynamically allocate an in-memory array
     // For now require a max returned parameter to pre-allocate and then shrink
-    // TODO ASv2 probably need a more gas-efficient lookup for the single most-recent attestation for one trusted issuer
+    // TODO ASv2 is it a risk to allocate an array of size maxAttestations?
+    // Same index corresponds to same attestation
+    address[] memory accounts = new address[](maxAttestations);
+    uint256[] memory issuedOns = new uint256[](maxAttestations);
+    address[] memory signers = new address[](maxAttestations);
+
     uint256 currIndex = 0;
-    IdentifierOwnershipAttestation[] memory attestations = new IdentifierOwnershipAttestation[](
-      maxAttestations
-    );
-    for (uint256 i = 0; i < trustedIssuers.length; i++) {
+
+    for (uint256 i = 0; i < trustedIssuers.length; i = i.add(1)) {
       address trustedIssuer = trustedIssuers[i];
-      for (uint256 j = 0; j < identifierToAddresses[identifier][trustedIssuer].length; j++) {
+      for (
+        uint256 j = 0;
+        j < identifierToAddresses[identifier][trustedIssuer].length;
```

Comment:
If `identifierToAddresses[identifier]` is undefined, indexing into `[trustedIssuer]` will throw an exception, right?

Reply:
I don't think so -- it looks like it'll return an empty mapping and then indexing into this second mapping returns an empty list (confirmed with test case). That said, definitely room to optimize the lookups here, will update